### PR TITLE
Fix signed <=> unsigned comparison warning

### PR DESCRIPTION
**Pull request by johnkingsley**
There are many ways to fix this, not sure what you prefer.
I can re-jig this patch if you prefer a different way.

### CODE REVIEW

File: src/ofxLua.h [hunk 1]
```diff
@@ -611,7 +611,7 @@ void ofxLua::writeVectorHelper(int type, vector<T>& v) {
 	clearTable();
 	
 	// add new variables
-	for(int i = 0; i < v.size(); ++i) {
+	for(typename vector<T>::size_type i = 0; i < v.size(); ++i) {
```
Comment (danomatika) [inline src/ofxLua.h]: Easiest fix is to cast to int:

    for(int i = 0; i < (int) v.size(); ++i) {

Comment (johnkingsley) [inline src/ofxLua.h]: OK, I will re-spin this patch

**Comment by johnkingsley** [comment]
OK, I updated the patch.  I also added changes to fix the same type of warnings in the luaExample.

**Comment by danomatika** [comment]
Do you get a warning with size_t or does that work as well. I'm thinking that using a true unsigned type is probably better than an int, although I highly doubt we'd run into type size issues.

Also, thanks for this. I'm mainly developing on macOS and clang is less pedantic about some warnings than gcc.

**Comment by johnkingsley** [comment]
`size_t` works fine (no warning).  This is on a raspberry pi.

The proper type is `vector<T>::size_type` which is almost always `typedef`'d to `size_t`.

Here are some links which discuss this issue:
http://stackoverflow.com/questions/14599057/how-compare-vector-size-with-an-integer
http://stackoverflow.com/questions/7636323/c-vector-size-types
http://stackoverflow.com/questions/4849632/vectorintsize-type-in-c

Think about what way you prefer and I'll re-do the patch as you like.

Thanks for ofxLua -- saved me a bunch of work!

**Comment by danomatika** [comment]
Does `auto` help in this case as well? I sometimes hate the pedantic-ness of some low level warnings like this so I'd prefer size_t over the explicit vector<T>::size_type...

**Comment by danomatika** [comment]
Also, how is the speed on RPI? What are you using ofxLua for? If you need an interpreter for just the OF bindings, I do have an app: http://danomatika.com/code/loaf

**Comment by johnkingsley** [comment]
I think the easiest, safest thing to do is use `size_t` as the type.
Using `auto` doesn't help when it's written `auto i = 0` since the the type will be `int`.

We could use a C++ iterator, something like:
```
for ( const auto & item : table ) {
        buffer << (string) item;
}
```
however there are cases where the we need to do something special
for the first element.  This complicates the above code.  I'm not sure
it's worth it.  Thoughts?

**Comment by johnkingsley** [comment]
Re: what I'm using ofxLua for...
https://github.com/johnkingsley/flexeplayer
It's for our local museum, to allow users to watch videos on a touchscreen computer.
It's currently a work-in-progress.  The OF part allows the user to pick the video.
The video is played using omxplayer.

So far, ofxLua works great. I probably would have started with loaf if I had of known about it first.
I think the extra thing I need to do is load some c++ code to handle the touchscreen.
I decided to do most of the coding in Lua since it's much faster developing in it
compared to c++ and I don't need the performance of c++.  My feeling was the
OF python interface wasn't as supported as the OF lua interface, so I went with lua.

As for performance, it works well on the raspberry pi 3. When no video is playing, load average is 0.1
and with a video playing it's 0.4

**Comment by danomatika** [comment]
Let's go with changing the iterators to size_t as it's the simplest and easiest to read solution.

> It's for our local museum, to allow users to watch videos on a touchscreen computer.

Nice to hear it's useful. I have a link to loaf on the readme but it's perhaps a bit small. I definitely agree that putting things together in Lua is both faster and more gratifying then C+. Also, I haven't seen any real speed difference between my stuff implemented in C++ and re-implemented in Lua except for the fact I was able to get it working in half the time.

As for additional C++ code, you can whip up your own bindings pretty easily with SWIG and write a makefile to compile them as a dynamic Lua module which can then be loaded into a Lua app via require. I have some tests from a while ago that work, but haven't put them online yet. I can put up the makefile if you want to try. This means the potential for being able to compile siddons and custom code as separate "plugins" for use with something like loaf or any kind of app using ofxLua.

**Comment by johnkingsley** [comment]
Ok, I've re-spun it using `size_t`
The warnings are gone and I tested luaExample on the raspberry pi
and it works AOK.

Re: using loaf... a dynamic loadable module is the way to go.
Right now I don't have time to re-jig my code. I need to do
some other things first -- my code goes live at the end of May!

**Pull request merged**